hazelcast/src/main/java/com/hazelcast/map/impl/operation/EntryOffloadableSetUnlockOperation.java

```diff
@@ -0,0 +1,262 @@
+
+import java.io.IOException;
+
+import static com.hazelcast.core.EntryEventType.ADDED;
+import static com.hazelcast.core.EntryEventType.REMOVED;
+import static com.hazelcast.core.EntryEventType.UPDATED;
+import static com.hazelcast.map.impl.EntryViews.createSimpleEntryView;
+import static com.hazelcast.map.impl.MapService.SERVICE_NAME;
+import static com.hazelcast.map.impl.recordstore.RecordStore.DEFAULT_TTL;
+
+/**
+ * Set & Unlock processing for the EntryOffloadableOperation
+ *
+ * See the javadoc on {@link EntryOffloadableOperation}
+ */
+public class EntryOffloadableSetUnlockOperation extends UnlockOperation {
+
+    protected String name;
+    protected Data value;
+    protected Data oldValue;
+    protected String caller;
+    protected long begin;
+    protected EntryEventType modificationType;
+
+    protected transient MapServiceContext mapServiceContext;
+    protected transient MapService mapService;
+    protected transient MapContainer mapContainer;
+    protected transient RecordStore recordStore;
+    protected transient MapEventPublisher mapEventPublisher;
+    protected transient Invalidator nearCacheInvalidator;
+
+    public EntryOffloadableSetUnlockOperation() {
+    }
+
+    public EntryOffloadableSetUnlockOperation(ObjectNamespace namespace, String name, EntryEventType modificationType,
+                                              Data key, Data oldValue, Data value, String caller, long threadId, long begin) {
+        super(namespace, key, threadId);
+        this.name = name;
+        this.value = value;
+        this.oldValue = oldValue;
+        this.caller = caller;
+        this.begin = begin;
+        this.modificationType = modificationType;
+    }
+
+    @Override
+    public void beforeRun() throws Exception {
+        super.beforeRun();
+        this.mapService = getNodeEngine().getService(SERVICE_NAME);
+        this.mapServiceContext = mapService.getMapServiceContext();
+        this.mapContainer = mapService.getMapServiceContext().getMapContainer(name);
+        this.mapEventPublisher = mapService.getMapServiceContext().getMapEventPublisher();
+        this.recordStore = mapServiceContext.getRecordStore(getPartitionId(), name);
+        this.nearCacheInvalidator = mapServiceContext.getMapNearCacheManager().getInvalidator();
+    }
+
+    @Override
+    public void run() throws Exception {
+        updateRecordStore();
+        unlockKey();
+    }
+
+    private void updateRecordStore() {
+        if (modificationType == REMOVED) {
+            recordStore.remove(key);
+            getLocalMapStats().incrementRemoves(getLatencyFrom(begin));
+        } else if (modificationType == ADDED || modificationType == UPDATED) {
+            recordStore.set(key, value, DEFAULT_TTL);
+            getLocalMapStats().incrementPuts(getLatencyFrom(begin));
+        } else {
+            throw new IllegalArgumentException("Unsupported event type " + modificationType);
+        }
+    }
+
+    private void unlockKey() {
```

Comment:
Thx. My concern was the duplication of the notified, but having implemented the change it looks better. Fixed.